apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php

```diff
@@ -147,4 +114,26 @@ public function enqueue_styles() {
 			filemtime( plugin_dir_path( __FILE__ ) . 'dist/' . $style_file )
 		);
 	}
+
+	/**
+	 * Fetch vertical data from the API or return cached version if available.
+	 */
+	public function fetch_vertical_data() {
+		$vertical_templates = get_transient( self::$transient_key );
+		$vertical_id        = get_site_option( 'site_vertical', 'default' );
+
+		// Load fresh data if we don't have any or vertical_id doesn't match.
+		if ( false === $vertical_templates || $vertical_id !== $vertical_templates['vertical_id'] ) {
+			$request_url = 'https://public-api.wordpress.com/wpcom/v2/verticals/' . $vertical_id . '/templates';
+			$response    = wp_remote_get( esc_url_raw( $request_url ) );
+			if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+				return false;
+			}
+			$vertical_templates                = json_decode( wp_remote_retrieve_body( $response ), true );
+			$vertical_templates['vertical_id'] = $vertical_id; // Add vertical_id so we can later compare it.
+			set_transient( self::$transient_key, $vertical_templates, 60 * 60 * 3 );
```

Comment:
3 hours are probably fine, I think a day wouldn't be too long either. Did you know about the handy [time constants](https://codex.wordpress.org/Transients_API#Using_Time_Constants) WP has?

Reply:
Now I do! changed to a day